### ISSUE TITLE
💅 `useNamingConvention`: allow consecutive capitals on generic parameters

### ISSUE THREAD

**Issue by kevinwolfcr** [biomejs/biome#2773]
### Environment information

```bash
CLI:
  Version:                      1.7.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.1.7"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Linter:
  Recommended:                  false
  All:                          true
  Rules:                        nursery/useSortedClasses = {"level":"error","options":{"attributes":["class","className","className","classList"],"functions":["clsx","cva","tw","cn"]}}

Workspace:
  Open Documents:               0
```

### Rule name

useNamingConvention

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZgB1AG4AYwB0AGkAbwBuACAAdABlAHMAdAA8AFQAQQByAGcAPgAoAGEAcgBnADoAIABUAEEAcgBnACkAIAB7AAoAIAAgAHIAZQB0AHUAcgBuACAAYQByAGcACgB9AAoACgB0AGUAcwB0ACgAMQApAA%3D%3D

### Expected result

It should not report consecutive capitals on generic parameters, since a lot of people likes using `T{GenericName}`.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
You can disable the [strictCase](https://biomejs.dev/linter/rules/use-naming-convention/#strictcase) option?

**Comment by atheck** [comment]
I have the same issue. Maybe when using a "match" regex the "strict" option should only check the captured part of the name, because that's what the "formats" use.

**Comment by Conaclos** [comment]
> I have the same issue. Maybe when using a "match" regex the "strict" option should only check the captured part of the name, because that's what the "formats" use.

This is already the current behavior. Closing because the issue is solved.
